src/sql/src/kafka_util.rs

```diff
@@ -47,10 +46,61 @@ generate_extracted_config!(
     ),
     (StatisticsIntervalMs, i32, Default(1_000)),
     (TopicMetadataRefreshIntervalMs, i32),
-    (TransactionTimeoutMs, i32)
+    (TransactionTimeoutMs, i32),
+    (StartTimeOffset, i64),
+    (StartOffset, KafkaStartOffset)
 );
 
-impl TryFrom<KafkaConfigOptionExtracted> for BTreeMap<String, StringOrSecret> {
+/// A fully specified lets of start offsets for a Kafka consumer.
+#[derive(Debug)]
+pub struct KafkaStartOffset(pub Vec<i64>);
+
+impl TryFromValue<Value> for KafkaStartOffset {
+    fn try_from_value(v: Value) -> Result<Self, PlanError> {
+        match v {
+            Value::Number(v) => Ok(Self(vec![v
```

Comment:
I would just ensure that they're all vecs rather than letting a number be a vec of one element

Reply:
That will be another breaking change, but sounds good!

Reply:
No one can have any of these options defined, right? Like there is no migration from the legacy with options to these, so we can come up with wholly new semantics. So don't think we need to worry about breaking changes, AFAIK.

Reply:
top commit now has this change!